simulation/dashboard.py

```diff
@@ -0,0 +1,290 @@
+            ),
+            dcc.Graph(id="message_timeline", figure=message_timeline_fig),
+            dcc.Graph(id="message_histogram", figure=messages_histogram),
+            html.Div(
+                style={
+                    "display": "flex",
+                    "flex-direction": "row",
+                    "justify-content": "center",
+                },
+                children=[
+                    dcc.Graph(
+                        id="failure_rate_per_status", figure=failure_rate_per_status_fig
+                    ),
+                    dcc.Graph(
+                        id="observed_failure_rate_per_status",
+                        figure=observed_failure_rate_per_status_fig,
+                    ),
+                ],
+            ),
+        ]
+    )
+
+    @app.callback(
+        [
+            dash.Output(component_id="message_timeline", component_property="figure"),
+            dash.Output(component_id="message_histogram", component_property="figure"),
+        ],
+        [
+            dash.Input(component_id="y-axis", component_property="value"),
+            dash.Input(component_id="runs-success", component_property="value"),
+            dash.Input(component_id="runs-list", component_property="value"),
+            dash.Input(component_id="types-list", component_property="value"),
+            dash.Input(component_id="failure-rates-range", component_property="value"),
+            dash.Input(
+                component_id="observed-failure-rates-range", component_property="value"
+            ),
+        ],
+    )
+    def update_figure(
+        selected_y_axis,
+        selected_runs_success,
+        selected_run_ids,
+        selected_types,
+        selected_failures,
+        selected_observed_failures,
+    ):
+        df = data
+        if selected_runs_success != "All":
+            df = df[df["status"] == selected_runs_success]
+        df = df[
+            df["failure_rate"].isin(
+                [
+                    i
+                    for i in failure_rates
+                    if i <= selected_failures[1] and i >= selected_failures[0]
+                ]
+            )
+        ]
+        df = df[
+            df["observed_failure_rate"].isin(
+                [
+                    i
+                    for i in observed_failure_rates
+                    if i <= selected_observed_failures[1]
+                    and i >= selected_observed_failures[0]
+                ]
+            )
+        ]
+        df = df[df["run_id"].isin(selected_run_ids)]
+        df = df[df["type"].isin(selected_types)]
+
+        new_message_timeline = px.scatter(
+            df,
+            x=selected_y_axis + "_time",
+            y=selected_y_axis + "_id",
+            color="type",
+            hover_name="type",
+        )
+        new_message_histogram = px.histogram(df, x=selected_y_axis + "_time")
+        return new_message_timeline, new_message_histogram
+
+    app.run_server(debug=True)
```

Comment:
I have several remarks/suggestions for the dashboard:
- You should separate it from the rest of the `simulation` file so you don't mix up js/python files, e.g. in a `simulation/dashboard` directory.
- You need to add a `requirements.txt` file so anyone could `pip install` the dependencies. A good practice is to create a virtual env, manually `pip install <package>` for each dependency, and then `pip freeze > requirements.txt`. 
- You should add a python linter, some parts are already quite hard to read, especially the layout creation. I suggest [black](https://github.com/psf/black) ; it works pretty well and is already used elsewhere in Cozy.
- Add a `dashboard` entry in the README to explain how to run it and what's the purpose